user-interface/src/components/Stopwatch/index.js

```diff
@@ -12,39 +10,37 @@ import Grid from 'material-ui/Grid';
 import Paper from 'material-ui/Paper';
 import Typography from 'material-ui/Typography';
 
+// Adds leading zeros to a number
+const pad = (number, length, ending) => (`0000${number}`).slice(-length) + ending;
+
 /**
- * Stopwatch component
+ * Stopwatch component 
  * Keeps track of elapsed mission time.
  */
 export default class Stopwatch extends Component {
-  constructor(props) {
-    super(props);
-
-    /*    time: the display string
-          startLabel: the label for the start button
-          startTime: when the time was last resumed/started
-          accTime: the time accumulated before the timer was paused
-          intervalId: the interval driving the stopwatch
-    */
-    this.state = { time: 'some time value', startLabel: 'Start', startTime: this.props.s || 0, accTime: 0, intervalId: null };
-    this.formatTime(); // TODO: format the time
-  }
-
-  // TODO: Clear interval on unmount!
-  formatTime() {
-    // Create a function to add leading zeros
-    function pad(number, length, ending) {
-      return (`0000${number}`).slice(-length) + ending;
-    }
+  /**
+   * state:
+   *  time: displayed time value (string)
+   *  active: whether it is running (boolean)
+   * startTime: when the time was last started/resumed (int)
+   * accTime: how much time has accumulated (int)
+   * intervalId: the interval updating the time
+   */
+  state = { time: '00:00:00.000', active: false }
+  startTime = 0
+  accTime = 0
+  intervalId = null
 
+  // Updates the display
+  formatTime = () => {
```

Comment:
We should try and refactor this function as just by reading the name we would assume it only does 1 thing: formats time. However, it does 2 things: formats time and modifies state. So we should break up this function into 2 parts: `formatTime` and `tick`. Where `tick` will modify the state.